### PR TITLE
adding few kafka and pulsar impls

### PR DESCRIPTION
**Pull request by shivasishdas**
Few challenges generalizing the interface to make it work for both Kafka and Pulsar clients.

- Kafka producer can accept Topic name at runtime to produce messages whereas Pulsar producers are inited using a specific Topic name
- Both consumers init have to be changed in order to init on-demand to consume on specific topics.

- Pulsar Admin APIs are not exposed in the go-client. Need to leverage pulsarctl impl for this else use the REST APIs directly https://streamnative.io/en/blog/tech/2019-11-26-introduction-pulsarctl

- Views around response generalization?

### CODE REVIEW

File: pkg/messagebroker/pulsar.go [hunk 1]
```diff
@@ -2,45 +2,106 @@ package messagebroker
...
+	Consumer pulsar.Consumer
+	Producer pulsar.Producer
 }
 
-// CreateTopic ...
-func (p PulsarBroker) CreateTopic(topic string) error {
-	//TODO: Implement me
-	return nil
+// NewPulsarBroker returns a pulsar broker
+func NewPulsarBroker(ctx context.Context, bConfig *BrokerConfig) (Broker, error) {
```
Comment (Abhishekvrshny) [inline pkg/messagebroker/pulsar.go:18]: same comment as that of kafka

File: pkg/messagebroker/kafka.go [hunk 2]
```diff
@@ -20,30 +20,44 @@ type KafkaBroker struct {
 
 // NewKafkaBroker returns a kafka broker
 func NewKafkaBroker(ctx context.Context, bConfig *BrokerConfig) (Broker, error) {
```
[the remark this answers is not in the document]
Comment (shivasishdas) [reply pkg/messagebroker/kafka.go:22]: Yeah, will break it. It makes more sense to have it separately.

File: pkg/messagebroker/pulsar.go [hunk 3]
```diff
@@ -3,44 +3,128 @@ package messagebroker
...
+	Ctx      context.Context
+	Config   *BrokerConfig
+	Consumer pulsar.Consumer
+	Producer pulsar.Producer
+}
+
+// NewPulsarConsumer returns a pulsar consumer
+func NewPulsarConsumer(ctx context.Context, bConfig *BrokerConfig) (Consumer, error) {
+	client, err := pulsar.NewClient(pulsar.ClientOptions{
+		URL:               bConfig.Brokers[0],
```
Comment (Abhishekvrshny) [inline pkg/messagebroker/pulsar.go:21]: put a validation saying pulsar broker list should have only 1 member during init

Comment (Abhishekvrshny) [reply pkg/messagebroker/pulsar.go:21]: check in registry

File: pkg/messagebroker/pulsar.go [hunk 4]
```diff
@@ -3,44 +3,128 @@ package messagebroker
...
+	panic("implement me")
 }
 
-// GetMessages returns a list of messages
-func (p PulsarBroker) GetMessages(numOfMessages int, timeout time.Duration) ([]string, error) {
-	//TODO: Implement me
-	return nil, nil
+// SendMessage sends a message on the topic
+func (p PulsarBroker) SendMessage(ctx context.Context, request SendMessageToTopicRequest) (SendMessageToTopicResponse, error) {
+	msgID, err := p.Producer.Send(context.Background(), &pulsar.ProducerMessage{
```
Comment (Abhishekvrshny) [inline pkg/messagebroker/pulsar.go:97]: use the same context

**Review by Abhishekvrshny — APPROVED** [review]

**Comment by Abhishekvrshny** [comment]
Add unit and integration tests in next PRs. Resolve the conflicts and merge.

**Pull request merged**